src/com/xilinx/rapidwright/rwroute/RoutableGraph.java

```diff
@@ -0,0 +1,150 @@
+package com.xilinx.rapidwright.rwroute;
+
+import com.xilinx.rapidwright.design.Net;
+import com.xilinx.rapidwright.device.Node;
+import com.xilinx.rapidwright.util.Pair;
+import com.xilinx.rapidwright.util.RuntimeTracker;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class RoutableGraph {
+
+    protected class RoutableNodeImpl extends RoutableNode {
+
+        public RoutableNodeImpl(Node node, RoutableType type) {
+            super(node, type);
+        }
+
+        @Override
+        protected Routable getOrCreate(Node node, RoutableType type) {
+            return RoutableGraph.this.getOrCreate(node, type).getFirst();
+        }
+
+        @Override
+        public boolean isExcluded(Node node) {
+            return preservedMap.containsKey(node) || super.isExcluded(node);
+        }
+
+        @Override
+        public Routable[] getChildren() {
+            setChildrenTimer.start();
+            setChildren();
+            setChildrenTimer.stop();
```

Comment:
```suggestion
            //setChildrenTimer.start();
            setChildren();
            //setChildrenTimer.stop();
```
Disabling these two lines improves runtime by 20%.  Should we remove or provide a disable flag to not call these by default? I've tried to disable the entire `RuntimeTracker` activities vs. just these two lines and the runtime is about the same.

Reply:
The only difference in reporting is that "rnodes creation" goes to 0:
```
Route design:                          162.13s
?? Initialization:                       0.56s
?  ?? determine route targets:           0.55s
?? Routing:                            161.57s
   ?? route clock:                       3.20s
   ?? route static nets:                 2.84s
   ?? route wire nets:                 152.78s
   ?  ?? rnodes creation:               30.00s
   ?  ?? route connections:            116.42s
   ?  ?? update timing:                  0.00s
   ?  ?? update congestion costs:        6.36s
   ?? finalize routes:                   2.76s
```

Reply:
I have a change that does that and retains the speedup.  I'm going to create it as a new PR so we can get it in sooner.  I'm assuming we might take a while with this one.

Reply:
This change depends on your branch, so I will push it here.